src/extension/infer.rs

```diff
@@ -323,10 +335,11 @@ impl UnificationContext {
         for tgt_node in hugr.nodes() {
             let sig: &OpType = hugr.get_nodetype(tgt_node).into();
             // Incoming ports with a dataflow edge
-            for port in hugr
-                .node_inputs(tgt_node)
-                .filter(|src_port| matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_))))
-            {
+            for port in hugr.node_inputs(tgt_node).filter(|src_port| {
+                matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_)))
```

Comment:
Yes, you *can* combine these into a single `matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_)) | Some(EdgeKind::Static(_)) | Some(EdgeKind::ControlFlow))`, I checked...